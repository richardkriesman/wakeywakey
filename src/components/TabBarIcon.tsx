import {Icon} from "expo";
import React, {ReactNode} from "react";

import Colors from "../constants/Colors";

export interface TabBarIconProps {
    name: string;
    focused: boolean;
}

export default class TabBarIcon extends React.Component<TabBarIconProps> {

    public render(): ReactNode {
        return (
            <Icon.Ionicons
                name={this.props.name}
                size={26}
                style={{marginBottom: -3}}
                color={this.props.focused ? Colors.tabIconSelected : Colors.tabIconDefault}
            />
        );
    }

}
