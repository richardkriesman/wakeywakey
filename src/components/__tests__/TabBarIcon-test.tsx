import * as React from "react";
import Renderer from "react-test-renderer";
import { TabBarIcon, TabBarIconProps } from "../TabBarIcon";

function createTestProps(focused: boolean): TabBarIconProps {
    return {
        focused,
        name: "ios-checkmark-circle-outline"
    };
}

describe("TabBarIcon", () => {

    it("matches snapshot focused", () => {
        const props = createTestProps(true);
        const json = Renderer.create(<TabBarIcon {...props} />).toJSON();
        expect(json).toMatchSnapshot();
    });

    it("matches snapshot unfocused", () => {
        const props = createTestProps(false);
        const json = Renderer.create(<TabBarIcon {...props} />).toJSON();
        expect(json).toMatchSnapshot();
    });

});
