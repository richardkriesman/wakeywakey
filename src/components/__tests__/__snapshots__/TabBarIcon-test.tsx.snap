// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarIcon matches snapshot focused 1`] = `<Text />`;

exports[`TabBarIcon matches snapshot unfocused 1`] = `<Text />`;
